code/modules/shuttle/shuttle.dm

```diff
@@ -0,0 +1,697 @@
+	if(M)
+		var/destination_found
+		for(var/obj/docking_port/stationary/S in shuttle_master.stationary)
+			if(!options.Find(S.id))
+				continue
+			if(M.canDock(S))
+				continue
+			destination_found = 1
+			dat += "<A href='?src=\ref[src];move=[S.id]'>Send to [S.name]</A><br>"
+		if(!destination_found)
+			dat += "<B>Shuttle Locked</B><br>"
+			if(admin_controlled)
+				dat += "Authorized personnel only<br>"
+				dat += "<A href='?src=\ref[src];request=1]'>Request Authorization</A><br>"
+	dat += "<a href='?src=\ref[user];mach_close=computer'>Close</a>"
+
+	var/datum/browser/popup = new(user, "computer", M ? M.name : "shuttle", 300, 200)
+	popup.set_content("<center>[dat]</center>")
+	popup.set_title_image(usr.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/shuttle/Topic(href, href_list)
+	if(..())
+		return
+	usr.set_machine(src)
+	src.add_fingerprint(usr)
+	if(!allowed(usr))
+		usr << "<span class='danger'>Access denied.</span>"
+		return
+
+	if(href_list["move"])
+		switch(shuttle_master.moveShuttle(shuttleId, href_list["move"], 1))
+			if(0)	usr << "<span class='notice'>Shuttle received message and will be sent shortly.</span>"
+			if(1)	usr << "<span class='warning'>Invalid shuttle requested.</span>"
+			else	usr << "<span class='notice'>Unable to comply.</span>"
+
+/obj/machinery/computer/shuttle/emag_act(mob/user)
+	if(!emagged)
+		src.req_access = list()
+		emagged = 1
+		user << "<span class='notice'>You fried the consoles ID checking system.</span>"
+
+/obj/machinery/computer/shuttle/ferry
+	name = "transport ferry console"
+	circuit = /obj/item/weapon/circuitboard/ferry
+	shuttleId = "ferry"
+	possible_destinations = "ferry_home;ferry_away"
+
+
+/obj/machinery/computer/shuttle/ferry/request
+	name = "ferry console"
+	circuit = /obj/item/weapon/circuitboard/ferry/request
+	var/cooldown //prevents spamming admins
+	possible_destinations = "ferry_home"
+	admin_controlled = 1
+
+/obj/machinery/computer/shuttle/ferry/request/Topic(href, href_list)
+	..()
+	if(href_list["request"])
+		if(cooldown)
+			return
+		cooldown = 1
+		usr << "<span class='notice'>Your request has been recieved by Centcom.</span>"
+		admins << "<b>FERRY: <font color='blue'>[key_name_admin(usr)] (<A HREF='?_src_=holder;adminmoreinfo=\ref[usr]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[usr]'>FLW</A>) (<A HREF='?_src_=holder;secretsfun=moveferry'>Move Ferry</a>)</b> is requesting to move the transport ferry to Centcom.</font>"
+		spawn(600) //One minute cooldown
+			cooldown = 0
+
+/obj/machinery/computer/shuttle/ert
+	name = "ert shuttle console"
+	//circuit = /obj/item/weapon/circuitboard/ert
+	shuttleId = "ert"
+	possible_destinations = "ert_home;ert_away"
+
+
+#undef DOCKING_PORT_HIGHLIGHT
+
+
+/turf/proc/copyTurf(turf/T)
+	if(T.type != type)
+		var/obj/O
+		if(underlays.len)	//we have underlays, which implies some sort of transparency, so we want to a snapshot of the previous turf as an underlay
```

Comment:
this doesn't actually really work correctly, the diagonal shuttle wall bits change to space regardless of what they are supposed to look like